synapse/media/media_repository.py

```diff
@@ -616,6 +616,7 @@ async def generate_local_exact_thumbnail(
                         height=t_height,
                         method=t_method,
                         type=t_type,
+                        length=t_byte_source.tell(),
```

Comment:
Sanity check: is `t_byte_source` definitely going to be fully written with the cursor at the end of the buffer, so that `.tell()` is the length? Both .crop() and .scale() say that they return "bytes ... ready to be written to disk" so I assume this is fine.

Reply:
My understand is that it will be written to and the cursor will be at the end of buffer; we could do something like:

```python
cur = t_byte_source.tell()
t_byte_source.seek(0, os.SEEK_END)
end = t_byte_source.tell()
t_byte_source.seek(cur, os.SEEK_SET)
```

If you're concerned about it not being at the end of the buffer.

Reply:
Reading back through the code I *think* this is fine TBH.

Reply:
That was my understanding too, I just wanted to check this through. Tinkering in an interpreter, our understanding seems to be correct:

```python
 20230728 (Red Hat 13.2.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from tests.test_utils import SMALL_PNG
>>> from io import BytesIO
>>> src = Image.open(BytesIO(SMALL_PNG))
>>> output = BytesIO()
>>> src.save(output, "png")
>>> output.tell()
70
>>> len(output.getvalue())
70
```

Though curiously this is slightly larger than the original PNG:
```python
>>> len(SMALL_PNG)
67
```

Reply:
Anyway LGTM, :shipit: